packages/react-router/modules/Prompt.js

```diff
@@ -39,9 +44,28 @@ class Prompt extends React.Component {
     }
   }
 
+  onbeforeunload(e) {
+    var dialogText = "Changes you made may not be saved."
+    e.returnValue = dialogText
+    return dialogText
+  }
+
+  enableUnload(_onbeforeunload) {
+    if(this._subscribed) { return }
+    this._subscribed = _onbeforeunload ? _onbeforeunload : this.onbeforeunload;
+    window.addEventListener(BEFOREUNLOAD, this._subscribed)
```

Comment:
Note that this behavior now makes this component dependent on the DOM, which I'd like to avoid for the generic `<Prompt>` component. Let's do this work in a separate `<BrowserPrompt>` component instead that wraps `<Prompt>`.